packages/gnome-shell/src/types/extension-metadata.ts

```diff
@@ -1,12 +1,17 @@
+import type Gio from "@girs/gio-2.0";
+
 export interface ExtensionMetadata {
-    uuid: string;
-    name?: string;
-    description?: string;
+    readonly uuid: string;
+    readonly dir: Gio.File,
+    readonly path: string,
+    readonly name: string;
+    readonly description: string;
     version?: string;
     url?: string;
-    "shell-version"?: string[];
+    readonly "shell-version": string[];
     "settings-schema"?: string;
     "gettext-domain"?: string;
     "original-author"?: string[];
     "extension-id"?: string;
-}
\ No newline at end of file
+    "version-name"?: string;
+}
```

Comment:
No reason to withheld the `readonly` modifier from the optional fields comes to mind. Have you withheld it on purpose from those fields?

Reply:
No particular reason; I just blindly made all properties readonly as documented in the guide and didn't care about the rest 😇 

I've made them all readonly.